### PR TITLE
ci: test ansible 2.5

### PR DESCRIPTION
**Pull request by leseb**
Signed-off-by: Sébastien Han <seb@redhat.com>

### CODE REVIEW

**Comment by leseb** [comment]
Closes: https://github.com/ceph/ceph-ansible/issues/2470

**Comment by andymcc** [comment]
Not that it's necessary to change anything, but the config_template plugin got split into it's own repo to separate it out from all the other OSA specific plugins:

https://github.com/openstack/ansible-config_template - right now it should be the same as the version from master in the openstack-ansible-plugins repo, but for future i think that will change.

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-docker_cluster

**Comment by leseb** [comment]
@andymcc thanks for the heads-up.

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-docker_cluster

**Comment by baoboa** [comment]
hello

In a redhat/centos7 with epel,centos-ceph-luminous repo ceph-ansible (3.1.0.0-0.beta6.1.el7) is broken, what is the plan ? to update to upstream version of ansible 2.5 or to provide ansible 2.4 package ? 

goodbye

**Comment by leseb** [comment]
@baoboa what's broken?

**Comment by baoboa** [comment]
@leseb 
#2475 is still pending if i am not mistaken

**Comment by andymcc** [comment]
@baoboa - mirroring the ansible-config_template from openstack would achieve the same result which is why I believe that PR #2475 is closed in favour of this.

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-centos7_cluster

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-docker_cluster

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-centos7_cluster

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-centos7_cluster

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-docker_cluster

**Comment by leseb** [comment]
Ansible 2.5 now fails when a key is missing where in 2.4 it wasn't complaining: 

```MSG:

the remote file does not exist, not transferring, ignored
```

Discovered a bug with that :)

**Comment by leseb** [comment]
Actually it's not a bug, the key doesn't exist yet, but the last mon will create it (this fails on mon0) so nothing is present (yet).

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-docker_cluster

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-centos7_cluster

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-update_cluster

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-purge_bluestore_osds_non_container

File: tox.ini [hunk 1]
```diff
@@ -161,8 +161,10 @@ deps=
   ansible2.2: -r{toxinidir}/tests/requirements2.2.txt
   ansible2.3: ansible==2.3.1
   ansible2.3: -r{toxinidir}/tests/requirements2.2.txt
-  ansible2.4: ansible==2.4.2
+  ansible2.4: ansible==2.4.1
```
Comment (guits) [inline tox.ini]: typo/leftover?

Comment (leseb) [reply tox.ini]: yes

**Pull request merged**